<?xml version="1.0" encoding="UTF-8"?>
<testng-results ignored="13" total="26" passed="12" failed="1" skipped="0">
  <reporter-output>
  </reporter-output>
  <suite started-at="2022-01-21T10:21:11 ICT" name="Suite" finished-at="2022-01-21T10:21:46 ICT" duration-ms="35157">
    <groups>
    </groups>
    <test started-at="2022-01-21T10:21:11 ICT" name="Test1" finished-at="2022-01-21T10:21:46 ICT" duration-ms="35157">
      <class name="tests.LoginTest">
        <test-method is-config="true" signature="setUp(java.lang.String)[pri:0, instance:tests.LoginTest@31fa1761]" started-at="2022-01-21T10:21:11 ICT" name="setUp" finished-at="2022-01-21T10:21:14 ICT" duration-ms="3275" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[test1]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- setUp -->
        <test-method signature="loginWithBlankField()[pri:0, instance:tests.LoginTest@31fa1761]" started-at="2022-01-21T10:21:14 ICT" name="loginWithBlankField" finished-at="2022-01-21T10:21:15 ICT" duration-ms="826" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- loginWithBlankField -->
        <test-method signature="loginByAdmin()[pri:0, instance:tests.LoginTest@31fa1761]" started-at="2022-01-21T10:21:15 ICT" name="loginByAdmin" finished-at="2022-01-21T10:21:25 ICT" duration-ms="10682" status="FAIL">
          <exception class="org.awaitility.core.ConditionTimeoutException">
            <message>
              <![CDATA[Condition with tests.LoginTest was not fulfilled within 10 seconds.]]>
            </message>
            <full-stacktrace>
              <![CDATA[org.awaitility.core.ConditionTimeoutException: Condition with tests.LoginTest was not fulfilled within 10 seconds.
at org.awaitility.core.ConditionAwaiter.await(ConditionAwaiter.java:165)
at org.awaitility.core.CallableCondition.await(CallableCondition.java:78)
at org.awaitility.core.CallableCondition.await(CallableCondition.java:26)
at org.awaitility.core.ConditionFactory.until(ConditionFactory.java:895)
at org.awaitility.core.ConditionFactory.until(ConditionFactory.java:864)
at tests.LoginTest.loginByAdmin(LoginTest.java:39)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:64)
at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
at java.base/java.lang.reflect.Method.invoke(Method.java:564)
at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:133)
at org.testng.internal.TestInvoker.invokeMethod(TestInvoker.java:598)
at org.testng.internal.TestInvoker.invokeTestMethod(TestInvoker.java:173)
at org.testng.internal.MethodRunner.runInSequence(MethodRunner.java:46)
at org.testng.internal.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:824)
at org.testng.internal.TestInvoker.invokeTestMethods(TestInvoker.java:146)
at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:146)
at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:128)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
at org.testng.TestRunner.privateRun(TestRunner.java:794)
at org.testng.TestRunner.run(TestRunner.java:596)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:377)
at org.testng.SuiteRunner.access$000(SuiteRunner.java:28)
at org.testng.SuiteRunner$SuiteWorker.run(SuiteRunner.java:418)
at org.testng.internal.thread.ThreadUtil.lambda$execute$0(ThreadUtil.java:64)
at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630)
at java.base/java.lang.Thread.run(Thread.java:832)
Caused by: java.lang.NullPointerException: Cannot invoke "org.openqa.selenium.WebDriver.getCurrentUrl()" because the return value of "tests.LoginTest.getDriver()" is null
at tests.LoginTest.lambda$0(LoginTest.java:39)
at org.awaitility.core.CallableCondition$ConditionEvaluationWrapper.eval(CallableCondition.java:99)
at org.awaitility.core.ConditionAwaiter$ConditionPoller.call(ConditionAwaiter.java:222)
at org.awaitility.core.ConditionAwaiter$ConditionPoller.call(ConditionAwaiter.java:209)
... 4 more
]]>
            </full-stacktrace>
          </exception> <!-- org.awaitility.core.ConditionTimeoutException -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- loginByAdmin -->
        <test-method is-config="true" signature="tearDown()[pri:0, instance:tests.LoginTest@31fa1761]" started-at="2022-01-21T10:21:45 ICT" name="tearDown" finished-at="2022-01-21T10:21:46 ICT" duration-ms="715" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- tearDown -->
      </class> <!-- tests.LoginTest -->
      <class name="tests.Cat_ProductManagementTest">
        <test-method signature="clickProductManagement()[pri:0, instance:tests.Cat_ProductManagementTest@1e1a0406]" started-at="2022-01-21T10:21:26 ICT" name="clickProductManagement" finished-at="2022-01-21T10:21:26 ICT" duration-ms="509" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- clickProductManagement -->
        <test-method signature="addProductFail1()[pri:0, instance:tests.Cat_ProductManagementTest@1e1a0406]" started-at="2022-01-21T10:21:26 ICT" name="addProductFail1" finished-at="2022-01-21T10:21:28 ICT" duration-ms="2309" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- addProductFail1 -->
        <test-method signature="addProductFail2()[pri:0, instance:tests.Cat_ProductManagementTest@1e1a0406]" started-at="2022-01-21T10:21:28 ICT" name="addProductFail2" finished-at="2022-01-21T10:21:30 ICT" duration-ms="1933" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- addProductFail2 -->
        <test-method signature="addProductFail3()[pri:0, instance:tests.Cat_ProductManagementTest@1e1a0406]" started-at="2022-01-21T10:21:30 ICT" name="addProductFail3" finished-at="2022-01-21T10:21:32 ICT" duration-ms="1887" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- addProductFail3 -->
        <test-method signature="addProductFail4()[pri:0, instance:tests.Cat_ProductManagementTest@1e1a0406]" started-at="2022-01-21T10:21:32 ICT" name="addProductFail4" finished-at="2022-01-21T10:21:34 ICT" duration-ms="1959" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- addProductFail4 -->
        <test-method signature="priceNotCorrect()[pri:0, instance:tests.Cat_ProductManagementTest@1e1a0406]" started-at="2022-01-21T10:21:34 ICT" name="priceNotCorrect" finished-at="2022-01-21T10:21:36 ICT" duration-ms="1848" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- priceNotCorrect -->
        <test-method signature="addProductSuccess()[pri:0, instance:tests.Cat_ProductManagementTest@1e1a0406]" started-at="2022-01-21T10:21:36 ICT" name="addProductSuccess" finished-at="2022-01-21T10:21:38 ICT" duration-ms="2238" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- addProductSuccess -->
        <test-method signature="productCodeError()[pri:0, instance:tests.Cat_ProductManagementTest@1e1a0406]" started-at="2022-01-21T10:21:38 ICT" name="productCodeError" finished-at="2022-01-21T10:21:41 ICT" duration-ms="2391" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- productCodeError -->
        <test-method signature="editProduct()[pri:0, instance:tests.Cat_ProductManagementTest@1e1a0406]" started-at="2022-01-21T10:21:41 ICT" name="editProduct" finished-at="2022-01-21T10:21:43 ICT" duration-ms="1908" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- editProduct -->
        <test-method signature="deleteProduct()[pri:0, instance:tests.Cat_ProductManagementTest@1e1a0406]" started-at="2022-01-21T10:21:43 ICT" name="deleteProduct" finished-at="2022-01-21T10:21:45 ICT" duration-ms="2478" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- deleteProduct -->
      </class> <!-- tests.Cat_ProductManagementTest -->
      <class name="tests.Cat_CustomerTest">
        <test-method signature="clickCategory()[pri:0, instance:tests.Cat_CustomerTest@23c30a20]" started-at="2022-01-21T10:21:25 ICT" name="clickCategory" finished-at="2022-01-21T10:21:26 ICT" duration-ms="103" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- clickCategory -->
      </class> <!-- tests.Cat_CustomerTest -->
    </test> <!-- Test1 -->
  </suite> <!-- Suite -->
</testng-results>
